File: monotonic-stack/next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        int[] answer = new int[nums1.length];
        Arrays.fill(answer, -1);
        for(int i = 0; i<nums1.length; i++){
            boolean meet = false;
            for(int element2: nums2){
                if(meet && element2 > nums1[i]){
                    answer[i] = element2;
                    break;
                }
                if(nums1[i] == element2){
                    meet = true;
                }
                
            }
        }

        return answer;

    }
}

File: stack/next-greater-element-i.py
class Solution(object):
    def nextGreaterElement(self, nums1, nums2):
        result = [-1]*len(nums2)
        stack = []
        for i in range(len(nums2)):
            while len(stack) > 0 and nums2[stack[-1]] < nums2[i]:
                index = stack.pop()
                result[index] = nums2[i]
            stack.append(i)
        ans = []
        print(result)
        for ele in nums1:
            ans.append(result[nums2.index(ele)])
        return ans